connectors/src/connectors/intercom/temporal/sync_help_center.ts

```diff
@@ -0,0 +1,383 @@
+    "[Intercom] Collection deleted."
+  );
+
+  // Then we call ourself recursively on the children collections
+  const childrenCollections = await IntercomCollection.findAll({
+    where: {
+      connectorId,
+      parentId: collectionId,
+    },
+  });
+  await Promise.all(
+    childrenCollections.map(async (collection) => {
+      await _deleteCollection({
+        connectorId,
+        dataSourceConfig,
+        collection,
+        loggerArgs,
+      });
+    })
+  );
+}
+
+/**
+ * Syncs a collection and its children (collection & articles) from the database and the core data source.
+ */
+async function _syncCollection({
+  connectorId,
+  intercomClient,
+  dataSourceConfig,
+  loggerArgs,
+  collection,
+  parents,
+}: {
+  connectorId: ModelId;
+  intercomClient: IntercomClient;
+  dataSourceConfig: DataSourceConfig;
+  collection: IntercomCollectionType;
+  loggerArgs: Record<string, string | number>;
+  parents: string[];
+}) {
+  // Sync the Collection
+  let collectionOnDb = await IntercomCollection.findOne({
+    where: {
+      connectorId,
+      collectionId: collection.id,
+    },
+  });
+
+  if (collectionOnDb) {
+    await collectionOnDb.update({
+      name: collection.name,
+      description: collection.description,
+    });
+  } else {
+    collectionOnDb = await IntercomCollection.create({
+      connectorId: connectorId,
+      collectionId: collection.id,
+      intercomWorkspaceId: collection.workspace_id,
+      helpCenterId: collection.help_center_id,
+      parentId: collection.parent_id,
+      name: collection.name,
+      description: collection.description,
+      url: collection.url,
+      permission: "read",
+    });
+  }
+
+  const collectionContent = await renderDocumentTitleAndContent({
+    dataSourceConfig,
+    title: collection.name,
+    content: { prefix: null, content: collection.description, sections: [] },
+    createdAt: new Date(collection.created_at),
+    updatedAt: new Date(collection.updated_at),
+  });
+  const collectionDsDocumentId = getHelpCenterCollectionDocumentId(
+    collection.workspace_id,
+    collection.id
+  );
+  await upsertToDatasource({
+    dataSourceConfig,
+    documentId: collectionDsDocumentId,
+    documentContent: collectionContent,
+    documentUrl: collection.url,
+    timestampMs: collection.updated_at,
+    tags: [`name:${collection.name}`, `lastUpdatedAt:${collection.updated_at}`],
+    parents,
+    retries: 3,
+    delayBetweenRetriesMs: 500,
+    loggerArgs: {
+      ...loggerArgs,
+      collectionId: collection.id,
+    },
+    upsertContext: {
+      sync_type: "batch",
+    },
+  });
+
+  // Sync the Collection's articles
+  const [childrenArticlesOnIntercom, childrenArticlesOnDb] = await Promise.all([
+    fetchIntercomArticles(intercomClient, collection.id),
+    IntercomArticle.findAll({
+      where: { connectorId, parentId: collection.id },
+    }),
+  ]);
+
+  const promises = childrenArticlesOnIntercom.map(async (articleOnIntercom) => {
```

Comment:
How big is that set in general?

If we are retrying the activity we could probably skip based on the last time we saw that document compared to the workflow start time?

Reply:
From the few help centers I checked (so def not a perfect metric) there's always less than 10 collections and each collections usually holds 5 to 30 articles. 
As it's fairly small I did not add such logic but we can definitely do it!